src/test/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandlerTest.java

```diff
@@ -0,0 +1,125 @@
+import org.ethereum.beacon.discovery.message.NodesMessage;
+import org.ethereum.beacon.discovery.schema.NodeRecord;
+import org.ethereum.beacon.discovery.schema.NodeRecordInfo;
+import org.ethereum.beacon.discovery.schema.NodeSession;
+import org.ethereum.beacon.discovery.storage.NodeTable;
+import org.ethereum.beacon.discovery.util.Functions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FindNodeResponseHandlerTest {
+  private static final Bytes PEER_ID = Bytes.fromHexString("0x1234567890ABCDEF");
+  private static final Bytes REQUEST_ID = Bytes.fromHexString("0x1234");
+  private final NodeSession session = mock(NodeSession.class);
+  private final NodeTable nodeTable = mock(NodeTable.class);
+
+  @BeforeEach
+  public void setUp() {
+    when(session.getNodeTable()).thenReturn(nodeTable);
+    when(session.getNodeId()).thenReturn(PEER_ID);
+  }
+
+  @Test
+  public void shouldAddReceivedRecordsToNodeTableButNotNodeBuckets() {
+    final NodeInfo nodeInfo = TestUtil.generateNode(9000);
+    final int distance = Functions.logDistance(PEER_ID, nodeInfo.getNodeRecord().getNodeId());
+    final FindNodeResponseHandler handler = new FindNodeResponseHandler(singletonList(distance));
+
+    final List<NodeRecord> records = singletonList(nodeInfo.getNodeRecord());
+    final NodesMessage message = new NodesMessage(REQUEST_ID, records.size(), records);
+    assertThat(handler.handleResponseMessage(message, session)).isTrue();
+
+    final NodeRecordInfo nodeRecordInfo = NodeRecordInfo.createDefault(nodeInfo.getNodeRecord());
+    verify(nodeTable).save(nodeRecordInfo);
+    verify(session, never()).putRecordInBucket(nodeRecordInfo);
+  }
+
+  @Test
+  public void shouldUpdateNodeRecordsWhenSeqNumHigher() {
+    final NodeInfo nodeInfo = TestUtil.generateNode(9000);
+    final NodeRecord originalRecord = nodeInfo.getNodeRecord();
+    when(nodeTable.getNode(originalRecord.getNodeId()))
+        .thenReturn(Optional.of(NodeRecordInfo.createDefault(originalRecord)));
+    final int distance = Functions.logDistance(PEER_ID, originalRecord.getNodeId());
+    final FindNodeResponseHandler handler = new FindNodeResponseHandler(singletonList(distance));
+
+    final NodeRecord updatedRecord =
+        originalRecord.withUpdatedCustomField(
+            "test", Bytes.fromHexString("0x8888"), nodeInfo.getPrivateKey());
+    final List<NodeRecord> records = singletonList(updatedRecord);
+    final NodesMessage message = new NodesMessage(REQUEST_ID, records.size(), records);
+    assertThat(handler.handleResponseMessage(message, session)).isTrue();
+
+    final NodeRecordInfo nodeRecordInfo = NodeRecordInfo.createDefault(updatedRecord);
+    verify(nodeTable).save(nodeRecordInfo);
+    verify(session, never()).updateNodeRecord(any());
+    verify(session, never()).putRecordInBucket(any());
+  }
+
+  @Test
+  public void shouldUpdateNodeRecordInSessionWhenSelfEnrReturnedWithHigherSeqNum() {
```

Comment:
`whenOwnEnrReturned`...

Reply:
I meant to just rename this but forgot to actually push the change.  Self ENR sounds weird at times but it is actually the term used for the node's local ENR in most places.